lib/main.js

```diff
@@ -841,6 +843,17 @@ exports.typeset = function (data,callback) {
   if (serverState == STATE.READY) StartQueue();
 }
 
+// main API, callback and promise compatible
+exports.typeset = function (data, callback) {
+    if (callback) cbTypeset(data, callback);
+    else return new Promise(function (resolve, reject) {
+        cbTypeset(data, function (output, input) {
+            if (output.errors) reject(output.errors);
```

Comment:
I would have said no since `undefinedCharError` is false by default (and if you switch it on, you likely want to reject the promise).

However, I have noticed that switching it to `true` has no effect for the promise-based API, e.g.,

```javascript
var mjAPI = require("./lib/main.js");
mjAPI.config({undefinedCharError: true});

mjAPI.typeset({
    math: "呵呵",
    format: "TeX",
    htmlNode: true,
    html: true,
    svg: true,
    equationNumbers: "AMS"
}, function (data) {
    console.log(data.errors); // has errors
});


mjAPI.typeset({math:'呵呵 ', html:true})
    .then((output, input) => console.log(output))
    .catch(errors => console.log("Errors:",errors)); //does not have errors
```

I don't understand what's going on here. (Hard errors work fine by the way, e.g., in the test file.)